### PR TITLE
CI: Test Python 3.10 on MacOS and Windows too

### PR DESCRIPTION
**Pull request by lithomas1**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

Blocker for MacOS/Windows Python 3.10 wheels. We probably want to migrate these onto Azure as soon as Python 3.10 is available there for load balancing reasons.

### CODE REVIEW

**Comment by lithomas1** [comment]
New Windows failures :(
```
Traceback (most recent call last):
  File "pandas\_libs\tslibs\tzconversion.pyx", line 383, in pandas._libs.tslibs.tzconversion.tz_convert_utc_to_tzlocal
    return _tz_convert_tzlocal_utc(utc_val, tz, to_utc=False, fold=fold)
  File "pandas\_libs\tslibs\tzconversion.pyx", line 594, in pandas._libs.tslibs.tzconversion._tz_convert_tzlocal_utc
    delta = _tzlocal_get_offset_components(val, tz, to_utc, fold)
  File "pandas\_libs\tslibs\tzconversion.pyx", line 554, in pandas._libs.tslibs.tzconversion._tzlocal_get_offset_components
    dt = dt.astimezone(tz)
  File "C:\hostedtoolcache\windows\Python\3.10.0-rc.2\x64\lib\site-packages\dateutil\tz\_common.py", line 144, in fromutc
    return f(self, dt)
  File "C:\hostedtoolcache\windows\Python\3.10.0-rc.2\x64\lib\site-packages\dateutil\tz\_common.py", line 261, in fromutc
    _fold = self._fold_status(dt, dt_wall)
  File "C:\hostedtoolcache\windows\Python\3.10.0-rc.2\x64\lib\site-packages\dateutil\tz\_common.py", line 196, in _fold_status
    if self.is_ambiguous(dt_wall):
  File "C:\hostedtoolcache\windows\Python\3.10.0-rc.2\x64\lib\site-packages\dateutil\tz\tz.py", line 254, in is_ambiguous
    naive_dst = self._naive_is_dst(dt)
  File "C:\hostedtoolcache\windows\Python\3.10.0-rc.2\x64\lib\site-packages\dateutil\tz\tz.py", line 260, in _naive_is_dst
    return time.localtime(timestamp + time.timezone).tm_isdst
OSError: [Errno 22] Invalid argument
```
No details because it raised in Cython code. I'll try to fix it.

File: .github/workflows/python-dev.yml [hunk 1]
```diff
@@ -40,12 +48,17 @@ jobs:
         python-version: '3.10-dev'
 
     - name: Install dependencies
+      shell: bash
       run: |
         python -m pip install --upgrade pip setuptools wheel
-        pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
-        pip install git+https://github.com/pytest-dev/pytest.git
+        echo $NUMPY_WHEELS_AVAILABLE
```
Comment (fangchenli) [inline .github/workflows/python-dev.yml:54]: Where do we set this variable?

Comment (lithomas1) [reply .github/workflows/python-dev.yml:54]: https://github.com/pandas-dev/pandas/pull/43772/files#diff-f6f33d5fc5cfd739f0d6efd63ea9dff886abe35cdfff5672e9a863ec8da8e72cR34

File: .github/workflows/python-dev.yml [hunk 2]
```diff
@@ -40,12 +48,16 @@ jobs:
...
-        pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
-        pip install git+https://github.com/pytest-dev/pytest.git
+        if [[ "$NUMPY_WHEELS_AVAILABLE" == "true" ]]; then
+          pip install -i https://pypi.anaconda.org/scipy-wheels-nightly/simple numpy
+        else
+          pip install git+https://github.com/numpy/numpy.git
+        fi
         pip install git+https://github.com/nedbat/coveragepy.git
-        pip install cython python-dateutil pytz hypothesis pytest-xdist pytest-cov
+        pip install cython python-dateutil pytz hypothesis pytest>=6.2.5 pytest-xdist pytest-cov
```
Comment (jreback) [inline .github/workflows/python-dev.yml:60]: is this in our requirements-dev?

**Review by jreback — CHANGES_REQUESTED** [review]

File: .github/workflows/python-dev.yml (hunk 2 above)
Comment (lithomas1) [reply .github/workflows/python-dev.yml:60]: Its 6.0 in requirements dev. 6.2.5 is the first version to support python 3.10

Comment (jreback) [reply .github/workflows/python-dev.yml:60]: oh ok got it, can you update whatsnew with this as well

**Comment by simonjayhawkins** [comment]
This is okay to backport?

**Comment by jreback** [comment]
yeah i think ok to backport

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @lithomas1

**Comment by jreback** [comment]
cc @lithomas1 if you'd do the manual backport when you have a chance

**Pull request merged**